sdks/go/pkg/beam/core/monitoring/monitoring.go

```diff
@@ -341,3 +343,189 @@ func int64Distribution(count, sum, min, max int64) ([]byte, error) {
 	}
 	return buf.Bytes(), nil
 }
+
+// FromMonitoringInfos extracts metrics from GetJobMetrics's response and
+// groups them into counters, distributions and gauges.
+func FromMonitoringInfos(attempted []*pipepb.MonitoringInfo, committed []*pipepb.MonitoringInfo) (
+	[]metrics.CounterResult,
+	[]metrics.DistributionResult,
+	[]metrics.GaugeResult) {
```

Comment:
Why not return a metrics.Results instead? Is the intent to force a breaking change if a new class of results is added?

Conversely, by using a metrics.Results, no such breaking change needs to happen, and things can change more gracefully (such as when generics land in Go).

Reply:
Good spot! Of course it should return `metrics.Results`. Thanks.